### PR TITLE
Converted escodegen.js to UMD format.

### LINKED ISSUE #115: Require.js compatibility
Is escodegen compatible with Require.js?  When required in, the generate method is undefined.

### PR DESCRIPTION
**Pull request by Victorystick**
Fixes #115.

### CODE REVIEW

File: escodegen.js [hunk 1]
```diff
@@ -2500,16 +2504,6 @@
         preserveBlankLines = options.format.preserveBlankLines && sourceCode !== null;
         extra = options;
 
-        if (sourceMap) {
```
Comment (michaelficarra) [inline escodegen.js]: The `sourceMap` here shadows the outer `sourceMap`, so this has changed behaviour. Use this instead:

```js
if (sourceMap) {
  SourceNode = sourceMap.SourceNode;
}
```

**Comment by Victorystick** [comment]
Alright. I'll fix that.

I also just realized that the `require("./package.json").version` got left behind. I'll get on that as well.

**Comment by Victorystick** [comment]
I get a single rather odd error. `source map test sourceContent support` failes with:
```
TypeError: undefined is not a function
      at Array.map (native)
      at Context.<anonymous> (test/source-map.js:503:24)
```
The Error is thrown on [this line](https://github.com/mozilla/source-map/blob/master/lib/source-map/source-map-consumer.js#L312). The sources variable is expected to be an array, but is an object when I run the tests.

**Comment by Victorystick** [comment]
This is not the way.

**Comment by roiDaniela** [comment]
Hi, I have same problem as mentiond here (not solved) - https://github.com/estools/escodegen/issues/300

Is there a fix? Thanx

**Pull request closed without merging**